cpp/open3d/data/Extract.h

```diff
@@ -0,0 +1,40 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// The MIT License (MIT)
+//
+// Copyright (c) 2018-2021 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <string>
+
+namespace open3d {
+namespace data {
+
+bool Extract(const std::string& filename,
```

Comment:
Let's refactor the interface. After reading more about "returning boolean" vs. "throw exception", I believe throwing exception is more appropriate here. 

- When an extraction fails, it is an "exceptional" case. 
- We'll return `true/false` when we expect the result to be either "true" or "false". In our case, we only expect the result to be "true". Otherwise, something unexpected happened. 
- In practice, we want to stop the execution and notify the user when a file fails to extract. 
- If we return true/false, we cannot make sure the developer always remember to do the true/false check. 

For the same reason, I am also changing the downloader APIs to throw exceptions rather than returning boolean flags. We can apply the same principle to other I/O functions.

Reply:
Done. [changed to `void` function, throwing `exception`].